pennylane/labs/trotter_error/fragments/vibrational_fragments.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The realspace vibrational Hamiltonian"""
+
+from typing import List, Sequence
+
+import numpy as np
+
+from pennylane.labs.trotter_error.realspace import RealspaceCoeffs, RealspaceOperator, RealspaceSum
+
+
+def vibrational_fragments(
+    modes: int, freqs: np.ndarray, taylor_coeffs: Sequence[np.ndarray], frag_method="harmonic"
+) -> List[RealspaceSum]:
+    """Returns a list of fragments summing to a vibrational Hamiltonian.
+
+    Args:
+        modes (int): the number of vibrational modes
+        freqs (ndarray): the harmonic frequences
+        taylor_coeffs (Sequence[ndarray]): a sequence containing the tensors of coefficients in the Taylor expansion
+        frag_method (string): the fragmentation method, valid options are ``harmonic``, ``kinetic``, and ``position``
+
+    Returns:
+        List[RealspaceSum]: a list of ``RealspaceSum`` objects representing the fragments of the vibrational Hamiltonian
+
+    **Example**
+
+    >>> from pennylane.labs.trotter_error import vibrational_fragments
+    >>> import numpy as np
+    >>> n_modes = 4
+    >>> freqs = np.random.random(4)
+    >>> taylor_coeffs = [np.array(0), np.random.random(size=(n_modes, )), np.random.random(size=(n_modes, n_modes))]
+    >>> vibrational_fragments(n_modes, freqs, taylor_coeffs)
```

Comment:
Might be better to print them as a loop so that it is easier to see the outputs.